src/EventListener/ProductTranslationUpdateListener.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\EventListener;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ORM\EntityManagerInterface;
+use Setono\SyliusRedirectPlugin\Factory\RedirectFactoryInterface;
+use Setono\SyliusRedirectPlugin\Finder\RemovableRedirectFinderInterface;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Sylius\Component\Product\Model\ProductInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final class ProductTranslationUpdateListener
```

Comment:
This class would be the extension point for other people wanting to redirect slugs. Therefore you should inject the class name and slug field in the constructor. This way you can make it dynamic so that it can be used with taxons and other resources having a slug

Reply:
Maybe the translation class should also be injected